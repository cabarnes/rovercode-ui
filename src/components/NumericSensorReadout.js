import React from 'react';
import { Typography, ExpansionPanelDetails } from '@material-ui/core';
import { withStyles } from '@material-ui/core/styles';
import { hot } from 'react-hot-loader';
import PropTypes from 'prop-types';

const styles = (theme) => ({
  expansionPanelSubDetails: {
    color: theme.palette.text.secondary,
    paddingTop: theme.spacing(0),
    paddingBottom: theme.spacing(1),
  },
});

const NumericSensorReadout = ({
  title,
  readings,
  unit,
  classes,
}) => (
  <>
    <ExpansionPanelDetails>
      <Typography>
        {title}
      </Typography>
    </ExpansionPanelDetails>
    {
      readings.map((reading) => (
        <ExpansionPanelDetails key={reading.label} className={classes.expansionPanelSubDetails}>
          <Typography variant="body2">
            {
              reading.reading === null ? (
                `${reading.label}: ? / ${reading.maxReading} ${unit}`
              ) : (
                `${reading.label}: ${reading.reading} / ${reading.maxReading} ${unit}`
              )
            }
          </Typography>
        </ExpansionPanelDetails>
      ))
    }
  </>
);

NumericSensorReadout.defaultProps = {
  unit: null,
};

NumericSensorReadout.propTypes = {
  classes: PropTypes.shape({
    expansionPanelSubDetails: PropTypes.string.isRequired,
  }).isRequired,
  title: PropTypes.string.isRequired,
  readings: PropTypes.arrayOf(PropTypes.shape({
    label: PropTypes.string.isRequired,
    reading: PropTypes.number,
    maxReading: PropTypes.number.isRequired,
  })).isRequired,
  unit: PropTypes.string,
};

export default hot(module)(withStyles(styles)(NumericSensorReadout));
