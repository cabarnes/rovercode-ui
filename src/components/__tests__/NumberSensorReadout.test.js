import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';

import NumericSensorReadout from '../NumericSensorReadout';

describe('The NumericSensorReadout component', () => {
  test('renders values on the page with no erros when values present', () => {
    const readings = [
      {
        label: 'A',
        reading: 42,
        maxReading: 1023,
      },
    ];

    const wrapper = mount(
      <NumericSensorReadout
        title="Flux Sensors"
        readings={readings}
        unit="Weber"
      />,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
    expect(wrapper.find('p').last().text()).toEqual('A: 42 / 1023 Weber');
  });

  test('renders on the page with no errors when no values', () => {
    const readings = [
      {
        label: 'A',
        reading: null,
        maxReading: 1023,
      },
    ];

    const wrapper = mount(
      <NumericSensorReadout
        title="Flux Sensors"
        readings={readings}
        unit="Weber"
      />,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
    expect(wrapper.find('p').last().text()).toEqual('Not connected');
  });
});
