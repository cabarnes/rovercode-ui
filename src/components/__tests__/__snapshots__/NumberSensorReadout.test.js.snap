// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The NumericSensorReadout component renders on the page with no errors when no values 1`] = `
<WithStyles(NumericSensorReadout)
  readings={
    Array [
      Object {
        "label": "A",
        "maxReading": 1023,
        "reading": null,
      },
    ]
  }
  title="Flux Sensors"
  unit="Weber"
>
  <NumericSensorReadout
    classes={
      Object {
        "expansionPanelSubDetails": "NumericSensorReadout-expansionPanelSubDetails-1",
      }
    }
    readings={
      Array [
        Object {
          "label": "A",
          "maxReading": 1023,
          "reading": null,
        },
      ]
    }
    title="Flux Sensors"
    unit="Weber"
  >
    <WithStyles(ForwardRef(ExpansionPanelDetails))>
      <ForwardRef(ExpansionPanelDetails)
        classes={
          Object {
            "root": "MuiExpansionPanelDetails-root",
          }
        }
      >
        <div
          className="MuiExpansionPanelDetails-root"
        >
          <WithStyles(ForwardRef(Typography))>
            <ForwardRef(Typography)
              classes={
                Object {
                  "alignCenter": "MuiTypography-alignCenter",
                  "alignJustify": "MuiTypography-alignJustify",
                  "alignLeft": "MuiTypography-alignLeft",
                  "alignRight": "MuiTypography-alignRight",
                  "body1": "MuiTypography-body1",
                  "body2": "MuiTypography-body2",
                  "button": "MuiTypography-button",
                  "caption": "MuiTypography-caption",
                  "colorError": "MuiTypography-colorError",
                  "colorInherit": "MuiTypography-colorInherit",
                  "colorPrimary": "MuiTypography-colorPrimary",
                  "colorSecondary": "MuiTypography-colorSecondary",
                  "colorTextPrimary": "MuiTypography-colorTextPrimary",
                  "colorTextSecondary": "MuiTypography-colorTextSecondary",
                  "displayBlock": "MuiTypography-displayBlock",
                  "displayInline": "MuiTypography-displayInline",
                  "gutterBottom": "MuiTypography-gutterBottom",
                  "h1": "MuiTypography-h1",
                  "h2": "MuiTypography-h2",
                  "h3": "MuiTypography-h3",
                  "h4": "MuiTypography-h4",
                  "h5": "MuiTypography-h5",
                  "h6": "MuiTypography-h6",
                  "noWrap": "MuiTypography-noWrap",
                  "overline": "MuiTypography-overline",
                  "paragraph": "MuiTypography-paragraph",
                  "root": "MuiTypography-root",
                  "srOnly": "MuiTypography-srOnly",
                  "subtitle1": "MuiTypography-subtitle1",
                  "subtitle2": "MuiTypography-subtitle2",
                }
              }
            >
              <p
                className="MuiTypography-root MuiTypography-body1"
              >
                Flux Sensors
              </p>
            </ForwardRef(Typography)>
          </WithStyles(ForwardRef(Typography))>
        </div>
      </ForwardRef(ExpansionPanelDetails)>
    </WithStyles(ForwardRef(ExpansionPanelDetails))>
    <WithStyles(ForwardRef(ExpansionPanelDetails))
      className="NumericSensorReadout-expansionPanelSubDetails-1"
      key="A"
    >
      <ForwardRef(ExpansionPanelDetails)
        className="NumericSensorReadout-expansionPanelSubDetails-1"
        classes={
          Object {
            "root": "MuiExpansionPanelDetails-root",
          }
        }
      >
        <div
          className="MuiExpansionPanelDetails-root NumericSensorReadout-expansionPanelSubDetails-1"
        >
          <WithStyles(ForwardRef(Typography))
            variant="body2"
          >
            <ForwardRef(Typography)
              classes={
                Object {
                  "alignCenter": "MuiTypography-alignCenter",
                  "alignJustify": "MuiTypography-alignJustify",
                  "alignLeft": "MuiTypography-alignLeft",
                  "alignRight": "MuiTypography-alignRight",
                  "body1": "MuiTypography-body1",
                  "body2": "MuiTypography-body2",
                  "button": "MuiTypography-button",
                  "caption": "MuiTypography-caption",
                  "colorError": "MuiTypography-colorError",
                  "colorInherit": "MuiTypography-colorInherit",
                  "colorPrimary": "MuiTypography-colorPrimary",
                  "colorSecondary": "MuiTypography-colorSecondary",
                  "colorTextPrimary": "MuiTypography-colorTextPrimary",
                  "colorTextSecondary": "MuiTypography-colorTextSecondary",
                  "displayBlock": "MuiTypography-displayBlock",
                  "displayInline": "MuiTypography-displayInline",
                  "gutterBottom": "MuiTypography-gutterBottom",
                  "h1": "MuiTypography-h1",
                  "h2": "MuiTypography-h2",
                  "h3": "MuiTypography-h3",
                  "h4": "MuiTypography-h4",
                  "h5": "MuiTypography-h5",
                  "h6": "MuiTypography-h6",
                  "noWrap": "MuiTypography-noWrap",
                  "overline": "MuiTypography-overline",
                  "paragraph": "MuiTypography-paragraph",
                  "root": "MuiTypography-root",
                  "srOnly": "MuiTypography-srOnly",
                  "subtitle1": "MuiTypography-subtitle1",
                  "subtitle2": "MuiTypography-subtitle2",
                }
              }
              variant="body2"
            >
              <p
                className="MuiTypography-root MuiTypography-body2"
              >
                A: ? / 1023 Weber
              </p>
            </ForwardRef(Typography)>
          </WithStyles(ForwardRef(Typography))>
        </div>
      </ForwardRef(ExpansionPanelDetails)>
    </WithStyles(ForwardRef(ExpansionPanelDetails))>
  </NumericSensorReadout>
</WithStyles(NumericSensorReadout)>
`;

exports[`The NumericSensorReadout component renders values on the page with no erros when values present 1`] = `
<WithStyles(NumericSensorReadout)
  readings={
    Array [
      Object {
        "label": "A",
        "maxReading": 1023,
        "reading": 42,
      },
    ]
  }
  title="Flux Sensors"
  unit="Weber"
>
  <NumericSensorReadout
    classes={
      Object {
        "expansionPanelSubDetails": "NumericSensorReadout-expansionPanelSubDetails-1",
      }
    }
    readings={
      Array [
        Object {
          "label": "A",
          "maxReading": 1023,
          "reading": 42,
        },
      ]
    }
    title="Flux Sensors"
    unit="Weber"
  >
    <WithStyles(ForwardRef(ExpansionPanelDetails))>
      <ForwardRef(ExpansionPanelDetails)
        classes={
          Object {
            "root": "MuiExpansionPanelDetails-root",
          }
        }
      >
        <div
          className="MuiExpansionPanelDetails-root"
        >
          <WithStyles(ForwardRef(Typography))>
            <ForwardRef(Typography)
              classes={
                Object {
                  "alignCenter": "MuiTypography-alignCenter",
                  "alignJustify": "MuiTypography-alignJustify",
                  "alignLeft": "MuiTypography-alignLeft",
                  "alignRight": "MuiTypography-alignRight",
                  "body1": "MuiTypography-body1",
                  "body2": "MuiTypography-body2",
                  "button": "MuiTypography-button",
                  "caption": "MuiTypography-caption",
                  "colorError": "MuiTypography-colorError",
                  "colorInherit": "MuiTypography-colorInherit",
                  "colorPrimary": "MuiTypography-colorPrimary",
                  "colorSecondary": "MuiTypography-colorSecondary",
                  "colorTextPrimary": "MuiTypography-colorTextPrimary",
                  "colorTextSecondary": "MuiTypography-colorTextSecondary",
                  "displayBlock": "MuiTypography-displayBlock",
                  "displayInline": "MuiTypography-displayInline",
                  "gutterBottom": "MuiTypography-gutterBottom",
                  "h1": "MuiTypography-h1",
                  "h2": "MuiTypography-h2",
                  "h3": "MuiTypography-h3",
                  "h4": "MuiTypography-h4",
                  "h5": "MuiTypography-h5",
                  "h6": "MuiTypography-h6",
                  "noWrap": "MuiTypography-noWrap",
                  "overline": "MuiTypography-overline",
                  "paragraph": "MuiTypography-paragraph",
                  "root": "MuiTypography-root",
                  "srOnly": "MuiTypography-srOnly",
                  "subtitle1": "MuiTypography-subtitle1",
                  "subtitle2": "MuiTypography-subtitle2",
                }
              }
            >
              <p
                className="MuiTypography-root MuiTypography-body1"
              >
                Flux Sensors
              </p>
            </ForwardRef(Typography)>
          </WithStyles(ForwardRef(Typography))>
        </div>
      </ForwardRef(ExpansionPanelDetails)>
    </WithStyles(ForwardRef(ExpansionPanelDetails))>
    <WithStyles(ForwardRef(ExpansionPanelDetails))
      className="NumericSensorReadout-expansionPanelSubDetails-1"
      key="A"
    >
      <ForwardRef(ExpansionPanelDetails)
        className="NumericSensorReadout-expansionPanelSubDetails-1"
        classes={
          Object {
            "root": "MuiExpansionPanelDetails-root",
          }
        }
      >
        <div
          className="MuiExpansionPanelDetails-root NumericSensorReadout-expansionPanelSubDetails-1"
        >
          <WithStyles(ForwardRef(Typography))
            variant="body2"
          >
            <ForwardRef(Typography)
              classes={
                Object {
                  "alignCenter": "MuiTypography-alignCenter",
                  "alignJustify": "MuiTypography-alignJustify",
                  "alignLeft": "MuiTypography-alignLeft",
                  "alignRight": "MuiTypography-alignRight",
                  "body1": "MuiTypography-body1",
                  "body2": "MuiTypography-body2",
                  "button": "MuiTypography-button",
                  "caption": "MuiTypography-caption",
                  "colorError": "MuiTypography-colorError",
                  "colorInherit": "MuiTypography-colorInherit",
                  "colorPrimary": "MuiTypography-colorPrimary",
                  "colorSecondary": "MuiTypography-colorSecondary",
                  "colorTextPrimary": "MuiTypography-colorTextPrimary",
                  "colorTextSecondary": "MuiTypography-colorTextSecondary",
                  "displayBlock": "MuiTypography-displayBlock",
                  "displayInline": "MuiTypography-displayInline",
                  "gutterBottom": "MuiTypography-gutterBottom",
                  "h1": "MuiTypography-h1",
                  "h2": "MuiTypography-h2",
                  "h3": "MuiTypography-h3",
                  "h4": "MuiTypography-h4",
                  "h5": "MuiTypography-h5",
                  "h6": "MuiTypography-h6",
                  "noWrap": "MuiTypography-noWrap",
                  "overline": "MuiTypography-overline",
                  "paragraph": "MuiTypography-paragraph",
                  "root": "MuiTypography-root",
                  "srOnly": "MuiTypography-srOnly",
                  "subtitle1": "MuiTypography-subtitle1",
                  "subtitle2": "MuiTypography-subtitle2",
                }
              }
              variant="body2"
            >
              <p
                className="MuiTypography-root MuiTypography-body2"
              >
                A: 42 / 1023 Weber
              </p>
            </ForwardRef(Typography)>
          </WithStyles(ForwardRef(Typography))>
        </div>
      </ForwardRef(ExpansionPanelDetails)>
    </WithStyles(ForwardRef(ExpansionPanelDetails))>
  </NumericSensorReadout>
</WithStyles(NumericSensorReadout)>
`;
